subsys/bluetooth/host/att.c

```diff
@@ -3310,6 +3317,25 @@ int bt_att_req_send(struct bt_conn *conn, struct bt_att_req *req)
 		return -ENOTCONN;
 	}
 
+#if defined(CONFIG_BT_GATT_CLIENT)
+	struct bt_att_hdr *hdr;
+
+	hdr = (void *)req->buf->data;
+
+	if (hdr->code == BT_ATT_OP_MTU_REQ) {
+		struct bt_att_chan *att_chan = att_get_fixed_chan(conn);
+
+		/* This request shall only be sent once during a connection by the client. */
+		if (atomic_test_bit(att_chan->flags, ATT_MTU_EXCHANGE_COMPLETE)) {
+			return -ECANCELED;
+		}
+
+		if (atomic_test_and_set_bit(att_chan->flags, ATT_MTU_EXCHANGE_IN_PROGRESS)) {
+			return -EALREADY;
+		}
+	}
+#endif /* CONFIG_BT_GATT_CLIENT */
+
```

Comment:
I'm a bit uncomfortable with special handling for a specific opcode in `bt_att_req_send`. It feel a bit arbitrarily placed here. I'm not sure enforcing the legality of sending a request should be the responsibility of `bt_att_req_send`.

Reply:
You have right, this is not a proper place for it. I will change that